### ISSUE TITLE
Problem installing updated version of Org

### ISSUE THREAD

**Issue by joelmccracken** [cask/cask#169]
This is a problem I have run into a few times:

Emacs now comes with a built-in version of Org. However, this is rather old, and I would like to use a more recent version.However,  I can't get cask to install any version of Org using the `depends-on` form. 

I _suspect_ that cask is detecting that org already exists in the load path and is giving up at that point.

IIRC, I have solved this in the past by bypassing Cask, but I would sure like to use cask to manage all of my package dependencies.

**Comment by alphapapa** [comment]
Hitting this issue myself, driving me a bit crazy.  My package requires Org 9, but the built-in version is 8.2.10.  No matter what I try, I can't get Cask to install a current version of Org, so my package's tests fail.

**Comment by NobbZ** [comment]
I do have the same problem, I can't convince cask to install a newer org-mode than what is available from the system-site…

The only way to circumvent this, seems to be to either directly pull from git-repo (I do not want to assume git everywhere) or by specifying `(depends-on "org-plus-contrib" "9.1.0")`.

Can't we have a `(depends-on)` that takes an elpa-repository as `:fetcher`? Or at least http/https to specify release tarballs from github?

**Comment by NobbZ** [comment]
See my transcript:

```plain
~/tmp/foo$ cask --version
0.8.1
~/tmp/foo$ emacs --version
GNU Emacs 24.5.1
Copyright (C) 2015 Free Software Foundation, Inc.
GNU Emacs comes with ABSOLUTELY NO WARRANTY.
You may redistribute copies of Emacs
under the terms of the GNU General Public License.
For more information about these matters, see the file named COPYING.
~/tmp/foo$ cask init
~/tmp/foo$ emacs Cask 
~/tmp/foo$ cat Cask
(source org)

(depends-on "org" "9.0.0")
~/tmp/foo$ cask install
~/tmp/foo$ cask eval '(message (org-version))'
8.2.10
~/tmp/foo$ emacs Cask
~/tmp/foo$ cask install
Making version-control local to org-plus-contrib-autoloads.el while let-bound!
Making generated-autoload-file local to  *autoload-file* while let-bound!
Wrote /home/nmelzer/tmp/foo/.cask/24.5/elpa/org-plus-contrib-20180108/org-plus-contrib-autoloads.el
Wrote /home/nmelzer/tmp/foo/.cask/24.5/elpa/org-plus-contrib-20180108/org-loaddefs.el
~/tmp/foo$ cask eval '(message (org-version))'
9.1.6
```

As you can see even while I have only `org` as repository and set the minimum version to `"9.0.0"` I get the built-in `"8.2.10"`. Only changing the dependency to `"org-plus-contrib"` makes cask able to install it.

**Comment by NobbZ** [comment]
Emacs version 25.3.1 yields the same.

**Comment by anquegi** [comment]
I also cannot install, I'm getting this error:
```

cask install                                                                                                                                                                                                       
error in process filter: Could not create connection to orgmode.org:443

```

I'm using mac os, it seems that my problem is the ssl conection, I tried with all this

```
(source org)
or
(source "org"       "http://orgmode.org/elpa/")
or
(source "org"       "https://orgmode.org/elpa/")
```

But nothing worked, I finally made this _silly_ thing:

I get the last two tars from here [https://orgmode.org/elpa/]( https://orgmode.org/elpa/)
 and put uncompressed folders in this folder:

/Users/toni/.emacs.d/.cask/25.3/elpa

```
$ cask eval '(message (org-version))'                                                                                                                                                                                
9.1.6
```

**Comment by sambrightman** [comment]
That looks like an unrelated SSL problem.

**Comment by anquegi** [comment]
That's true but in that case should work with the http route, and doesn't work, and @sambrightman, do you have any idea how to solve that issue?

**Comment by sambrightman** [comment]
I would use the `org-plus-contrib` workaround described by others (apparently this package is what most people want anyway).

**Comment by alphapapa** [comment]
> I would use the org-plus-contrib workaround described by others (apparently this package is what most people want anyway).

I hope that Org eventually drops the distinction and either makes that the `org` package, or figures out a way to install contrib separately.

If and when that happens, the workaround won't work.

**Comment by sambrightman** [comment]
Clearly. Hopefully this will issue will be resolved by then.

**Comment by sambrightman** [comment]
It seems I mis-remembered or mis-construed the version specifier working for me in the past. I recall getting a more modern flycheck via my Cask file. I also thought that comments above stated it was working for others but they are referring to the workaround. Per https://github.com/cask/cask/issues/119#issuecomment-25519384 it only ever worked for package building - the code also seems to show no history of it working for installation (making [the documentation](http://cask.readthedocs.io/en/latest/guide/dsl.html#el.function.depends-on) wrong too).

This should be fixed by #232, which is blocked on cask/epl#27.

**Comment by anquegi** [comment]
In my case the problem was the gnutls apparently installing emacs with:

`brew cask install emacs`, doesn't get well dependencies with gnutls

I installed with:

`brew install emacs --HEAD --with-cocoa --with-gnutls --with-librsvg --with-imagemagick@6 --with-mailutils`

and it works perfectly,

**Comment by alphapapa** [comment]
I do hope this issue can be resolved someday.  I thought that I could lock Org to a specific version by doing:

```el
(depends-on "org"
             :git "https://code.orgmode.org/bzg/org-mode.git"
             :ref "732aa477b031772c3694023cbf34eec6bdc08d61")

But after running `cask install` and `cask update`, running `cask exec` still uses the built-in, system version of Org.

So I tried using:
```
(depends-on "org-plus-contrib"
             :git "https://code.orgmode.org/bzg/org-mode.git"
             :ref "732aa477b031772c3694023cbf34eec6bdc08d61")
```

But of course that fails, because:
```
$ cask update
Wrong type argument: cask-dependency, void-function
```
and:
```
$ cask install
Dependency org-plus-contrib failed to install: Symbol’s function definition is void: package-build-package
Output:
```
So it seems that the only thing I can do is:
```el
(depends-on "org-plus-contrib" "9.0")
```
Which currently gets me Org `9.1.13`.  And sometimes that's fine, but it means that when Org pushes out an update, and the Cask version gets updated, it may break my tests (e.g. commits b5c19643d2347e4e1f4d76634b458380a46e39c4 and 00f95cb126af6239cf0f18a4db21bd47614f6894, see https://lists.gnu.org/archive/html/emacs-orgmode/2017-03/msg00361.html).  And when the new version of Org is different from the one I have installed in my personal Emacs config, it means that the tests work in interactive Emacs and fail in Cask, which is extra confusing.  (I've already spent a few hours on this tonight.)

**Comment by dickmao** [comment]
Fixes #203

**Comment by dickmao** [comment]
Consolidating with #203
